src/seedu/addressbook/data/person/Block.java

```diff
@@ -0,0 +1,55 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's address in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ */
+public class Block {
+
+    public static final String EXAMPLE = "123";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Block can be in any format";
+    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+
+    public final String value;
+    private boolean isPrivate;
+
+    /**
+     * Validates given address.
+     *
+     * @throws IllegalValueException if given address string is invalid.
+     */
+    public Block(String block) throws IllegalValueException {
```

Comment:
This function does not throw IllegalValueException. Add throws clause only when an exception is being thrown.